### DISCUSSION TITLE
ssd1306 on esp32 s3 dev kit

### DISCUSSION THREAD

**Discussion by G5OYY-uk** [BitMaker-hub/NerdMiner_v2#528] [Q&A]
has anyone successfully added a ssd1306 to a s3 dev board have been trying to figure this out all weekend but not got anywhere and head is spinning  just want to be able to monitor a board without having to fire up pc every time

**Comment by paelzer** [comment]
SSD1306 128x64 works fine on the ESP32-S3 devkitc 1, but the NerdminerV2 source code doesn't support i2c displays by default.

At least on the mentioned ESP32-S3 devkitc 1 I could implement this with these libraries by Adafruit:

https://github.com/adafruit/Adafruit_SSD1306
https://github.com/adafruit/Adafruit-GFX-Library

Pins used:
GPIO 8 (SDA)
GPIO 9 (SCL)
3.3V
GND

You can have a look here: https://github.com/paelzer/NerdMiner_v2_paelzer/tree/SSD1306
Check env:esp32_s3_devkitc_1_SSD1306

![image](https://github.com/user-attachments/assets/8a836d20-5613-4384-8929-cd1061fc2cbc)

**Comment by G5OYY-uk** [comment]
Got working after much messing however just been given a st7789 tft so.may
use that instead

**Reply by G5OYY-uk** [reply]
I can't believe it is that hard to get a display to work with these boards once you have navigated the confusing labeling of various boards I.e MOSI SD and various others the damn thing still does not work. Is there any definitive guide to setting up a st7789 with a S3 devkit c1 and maybe a clue as to.what needs to be altered on the code to get.to.work??? The S3 T Display works perfect on the board as far as the software goes but for the life.of me I can't alter it to run the external.st7789 even though it's the same controller

**Comment by paelzer** [comment]
Can you post more details to the ST7789 that you want to use? Maybe photos and exact naming of the device? I got the Nerdminer to work with many different types on ESP32 devkits and also ESP32 S3 devkits.

**Comment by G5OYY-uk** [comment]
I have a tenstar robo 2.4 240x320 st7789 driver screen and a Estardyn 1.9 240x240 tft which is also a st7789 driver and I am using a esp 32 s3 devkit c1 clone

**Comment by paelzer** [comment]
This one?

![image](https://github.com/user-attachments/assets/10b828e9-135f-4c1c-847d-073f86a0720d)

**Comment by G5OYY-uk** [comment]
I have these ones 
![Screenshot_20241202_142543_AliExpress](https://github.com/user-attachments/assets/b90369cb-295c-4d5a-8f6a-32d2d82abfd0)
![Screenshot_20241202_142513_AliExpress](https://github.com/user-attachments/assets/343d89d9-eb78-4fe4-9a03-4421fc89f3b0)
![Screenshot_20241202_142425_AliExpress](https://github.com/user-attachments/assets/15f3b0c5-1197-4d9a-99ed-ea8ffe9e5440)

**Comment by G5OYY-uk** [comment]
A d this controller 
![17331597495194503659819618609495](https://github.com/user-attachments/assets/aa6b212c-b578-46d3-a986-a5df63edc9db)
![1733159779441399371341904660396](https://github.com/user-attachments/assets/41ce41ce-8f4b-4a83-add8-2a3a9bd42a6c)

**Comment by paelzer** [comment]
For the ST7735 on the ESP32-S3 devkit I recommend this repo:

https://github.com/matlen67/NerdMiner_V2_ESP32-ST7735
In the platformio.ini, to have support for ESP32-S3 devkit, replace the config for env:ESP32_ST7735 with following:

[env:ESP32_ST7735]
platform = espressif32@6.6.0
board = esp32-s3-devkitc-1
framework = arduino
monitor_filters = 
    esp32_exception_decoder
    time
    log2file
board_build.arduino.memory_type = qio_opi
monitor_speed = 115200
;upload_speed = 115200
board_build.partitions = huge_app.csv
build_flags = 
    -D BOARD_HAS_PSRAM
    -D ARDUINO_USB_MODE=1
    -D ARDUINO_USB_CDC_ON_BOOT=1
    -D DEBUG_MINING=1
	-D ESP32_ST7735=1
  	-D TFT_RGB_ORDER=TFT_RGB
	-DUSER_SETUP_LOADED=1
	-DST7735_DRIVER=1
	-DTFT_WIDTH=128
	-DTFT_HEIGHT=160
	-DTFT_MOSI=11
	-DTFT_SCLK=12
	-DTFT_CS=10
	-DTFT_DC=2
	-DTFT_RST=4
	-DLOAD_GLCD=1
	-DLOAD_FONT2=1
	-DLOAD_GFXFF=1
	-DSMOOTH_FONT=1
	-DSPI_FREQUENCY=27000000
	-DSPI_READ_FREQUENCY=20000000
lib_deps = 
	https://github.com/takkaO/OpenFontRender
	bblanchon/ArduinoJson@^6.21.2
	https://github.com/tzapu/WiFiManager.git#v2.0.16-rc.2
	mathertel/OneButton @ ^2.0.3
	arduino-libraries/NTPClient
	bodmer/TFT_eSPI @ ^2.5.31
lib_ignore = 
	HANSOLOminerv2

Works fine here.

**Comment by paelzer** [comment]
Use this pinout:

GND
3.3V
SCK=12
SDA=11
RST=4
DC/RS=2
CS=10

**Comment by paelzer** [comment]
![image](https://github.com/user-attachments/assets/71f2eae1-76ad-4e49-b650-3d1f91ef9e7e)

**Comment by G5OYY-uk** [comment]
Thank you
I will try this a little.latee however I have both DC and CS pins donthese
gontonsame.pin or different ?

**Comment by G5OYY-uk** [comment]
Sorry just see it was RS not CS Doh!!!!

**Reply by paelzer** [reply]
CS, DC (sometimes also mentioned as RS) need to have different pins.

CS is the SPI slave select pin which is activated by the master if he wants to control the slave (the display in this case).

DC/RS pin is the display control pin (data/command).

**Comment by G5OYY-uk** [comment]
Great i finally jabe managed to get the sd1306 working as well as one of
the tft displays.wprking perfect the other 2 do not display properly will
have to play around with those and see if i can figure out the settings but
I would.never have managed this without your help thank you so much

**Reply by G5OYY-uk** [reply]
ok i have been playing a lot with this now and have managed to get several different screens working finally figured most of the problems were caused by me however my final one is a real humdinger lol 
the RGB led onboard will work if i flash the board as Lillygo s3 Amoled or another S3 version cant remember off top of head which but any other version does not want to know i wanted to break the LED to front of 3d case with the screen more for oooo twinkly led than anything else however after playing for several hours i find that the FastLED library does not seem to work with the S3 at all even flashing the board with exammple files will not work 
has anyone actually got round this issue ????
